src/cmd/cli/command/compose.go

```diff
@@ -338,8 +338,16 @@ func makeComposeDownCmd() *cobra.Command {
 				Verbose:            verbose,
 				LogType:            logs.LogTypeAll,
 			}
-			err = cli.Tail(cmd.Context(), provider, projectName, tailOptions)
+			tailCtx := cmd.Context()
+			err = cli.Tail(tailCtx, provider, projectName, tailOptions)
 			if err != nil {
+				if connect.CodeOf(err) == connect.CodePermissionDenied {
+					// If tail fails because of missing permission, we wait for the deployment to finish
+					term.Warn("Unable to tail logs. Waiting for down to finish.")
+					<-tailCtx.Done()
```

Comment:
this waits indefinitely

Reply:
Alternatively, we can treat missing permission as implied `--detach`.

Reply:
> Alternatively, we can treat missing permission as implied `--detach`.

That's an elegant idea!

Reply:
ce6aec5d